library/alloc/src/collections/btree/map.rs

```diff
@@ -1144,13 +1145,22 @@ impl<K, V> BTreeMap<K, V> {
     where
         K: Ord,
     {
-        // FIXME(@porglezomp) Avoid allocating if we don't insert
         let (map, dormant_map) = DormantMutRef::new(self);
-        let root_node = Self::ensure_is_owned(&mut map.root).borrow_mut();
+        let root_node = match map.root.as_mut() {
+            Some(root) => root.borrow_mut(),
```

Comment:
I really favour nesting the rest of the expression in here, and avoid the `return`  - whether it's before or after the `None` case.